output/baseoutput/clientsession.go

```diff
@@ -20,15 +23,25 @@ type clientSession struct {
 	onChunkAcked func(chunk base.LogChunk)
 	metrics      clientMetrics
 	conn         ClientConnection
-	leftovers    chan base.LogChunk        // unprocessed buffers from previous session(s)
-	lastChunk    *base.LogChunk            // last buffer in processing (to be added to leftovers if not completed)
-	ackerChan    chan base.LogChunk        // channel to pass buffers for acknowledger (wait for ACK and delete), close to end acknowledger
-	ackerQuit    *channels.SignalAwaitable // channel for acknowledger to signal its end
-	unacked      atomic.Value              // *[]base.LogChunk, un-ACK'ed buffers set when acknowledger quits (to be resent in next session)
+	leftovers    chan base.LogChunk        // unprocessed chunks from previous session(s)
+	lastChunk    *base.LogChunk            // last chunk in processing (to be added to leftovers if not completed)
+	ackerChan    chan base.LogChunk        // channel to pass chunks for acknowledger (wait for ACK and delete), close to end acknowledger
+	ackerQuit    *channels.SignalAwaitable // signal of acknowledger's end
+	unacked      atomic.Value              // *[]base.LogChunk, un-ACK'ed chunks set when acknowledger quits (to be resent in next session)
 }
 
+type reconnectPolicy string
+type acknowledgerEnding bool
```

Comment:
Maybe this should be string as well, to match the `reconnectPolicy` type. There isn't really any reason for it to be a boolean and it's a bit jarring to see both strings and booleans used for the same purpose.